pkg/nsx/services/subnet/builder.go

```diff
@@ -113,6 +113,15 @@ func (service *SubnetService) buildSubnetDHCPConfig(mode string) *model.SubnetDh
 	return subnetDhcpConfig
 }
 
+func (service *SubnetService) updateSubnetDHCPConfig(subnetDHCPConfig *model.SubnetDhcpConfig, mode string) *model.SubnetDhcpConfig {
+	nsxMode := nsxutil.ParseDHCPMode(mode)
+	subnetDhcpConfig := &model.SubnetDhcpConfig{
+		DhcpServerAdditionalConfig: subnetDHCPConfig.DhcpServerAdditionalConfig,
+		Mode:                       &nsxMode,
+	}
+	return subnetDhcpConfig
```

Comment:
since the subnetDHCPConfig is pointer, could we just update the subnetDHCPConfig.Mode instead of create a new one and return? it seems no need to return it

Reply:
updatedSubnet.SubnetDhcpConfig is a pointer to vpcSubnets[i].SubnetDhcpConfig before the modification. If we change the Mode on it, the vpcSubnets[i].SubnetDhcpConfig.Mode will also be modified and result in the content in store updated before updating success. So I need to create a new subnetDHCPConfig instead of updating it.